src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -126,31 +125,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   /**
-   * Creates a calendar event for each task with the same title, description and
-   * start time. The duration is the default one.
+   * Extends the tasks into a wrapper class that contains also the duration.
+   */
+  List<ExtendedTask> createExtendedTasks(List<Task> tasks, String[] tasksDurations) {
+    List<ExtendedTask> extendedTasks = new ArrayList<>();
+
+    for (ListIterator<Task> tasksIterator = tasks.listIterator(); tasksIterator.hasNext(); ) {
+      int index = tasksIterator.nextIndex();
+      ExtendedTask extendedTask = new ExtendedTask(tasksIterator.next(), tasksDurations[index]);
+      extendedTasks.add(extendedTask);
+    }
+
+    return extendedTasks;
+  }
+
+  /**
+   * Creates a calendar event for each extended task with the same title, description,
+   * duration and start time.
    */
-  List<Event> createEventsFromTasks(List<Task> tasks, String timeZone) {
+  List<Event> createEventsFromExtendedTasks(List<ExtendedTask> extendedTasks, String timeZone) {
```

Comment:
nit: This has been like this already but is a sign of _Primitive Obsession_. The time zone here and in other places should instead be a [`java.util.TimeZone`][1] object.

[1]: https://docs.oracle.com/javase/8/docs/api/java/util/TimeZone.html

Reply:
Once I was thinking about refactoring it, but then we needed the String when we created the events, and for some reason it seemed difficult to get back the string from the TimeZone.

Reply:
Yeah, the creation of an `EventDateTime` requires the time zone as a `String`, so especially here that is creating a new event is useful to have the time zone parameter as a `String`.

Reply:
Likely TimeZone#getID() would return the string you would need to pass